src/download-database.ts

```diff
@@ -0,0 +1,187 @@
+// This file borrows heavily from the actions "downloadTool" function:
+// https://github.com/actions/toolkit/blob/main/packages/tool-cache/src/tool-cache.ts#L38
+
+import * as fs from "fs";
+import * as path from "path";
+import * as stream from "stream";
+import * as util from "util";
+
+import * as core from "@actions/core";
+import * as httpm from "@actions/http-client";
+import { IHeaders } from "@actions/http-client/interfaces";
+import * as io from "@actions/io";
+
+export class HTTPError extends Error {
+  httpStatusCode: number | undefined;
+  httpMessage: string;
+  constructor(httpStatusCode: number | undefined, httpMessage: string) {
+    super(`Unexpected HTTP response: ${httpStatusCode}. ${httpMessage}`);
+    // Set status code and error message to `this`
+    this.httpStatusCode = httpStatusCode;
+    this.httpMessage = httpMessage;
+  }
+}
+
+const userAgent = "actions/tool-cache";
```

Comment:
What happens if we change this to "GitHub remote queries" or something similar. Probably shouldn't pretend to be the tool cache. The user agent is pretty much free text as I understand it so can put whatever we like.

Reply:
Thanks! That worked 🎉